### PR TITLE
Remove mutable parameters in provider api scripts

### LINKED ISSUE #99: [Quality] Remove mutable defaults in provider API scripts
## Current Situation

Provider API scripts use dictionaries as parameters, mainly for query parameters and headers. This is a problem because in Python, function parameters are initialized once. If several function calls make changes to the parameters, the parameter used is mutated. So, if you have an empty dictionary as a parameter, and add a key/value pair in one of the calls, you might expect that the default for the next function call to be the empty dictionary. However, it will reuse the previously created default object: a dictionary with the key/value pair added in the previous call.

## Suggested Improvement

We should replace the mutable defaults with `None`. If the default is None, we would set the parameter to the global variable value that we have.

## Benefit

Mutable defaults might cause difficult to debug errors. This change will prevent this kind of errors.

### PR DESCRIPTION
**Pull request by obulat**
Fixes #99 

This PR replaces mutable default parameters in provider api scripts.
This will eliminate one type of possible errors: when the default parameters change between the calls.

I split PR #67 into several PRs to make them more manageable and make reviews easier.

Signed-off-by: Olga Bulat <obulat@gmail.com>

### CODE REVIEW

**Review by krysal — COMMENTED** [review]
Splitting the PRs is appreciated. I was thinking it would be good to also add `.copy()` when assigning constant dictionaries to prevent mutations completely. What do you think?

File: src/cc_catalog_airflow/dags/provider_api_scripts/test_smithsonian.py [hunk 1]
```diff
@@ -11,7 +11,6 @@
 
 logging.basicConfig(
     format='{asctime} - {name} - {levelname}:  {message}',
-    style='{',
```
[the remark this answers is not in the document]
Comment (obulat) [reply src/cc_catalog_airflow/dags/provider_api_scripts/test_smithsonian.py:14]: Thank you for catching this! In other files, the format used for logger messages setup is `%(some_variable)`.
It is possible to set up logging using the `{some_variable}` setup with the added `style='{'` setting. Pycharm incorrectly marks this as an error, so I just removed this line. However, simply removing it causes this script to end with an error.
I've unified all logger setups to just use `%(some_variable)` setup.

File: src/cc_catalog_airflow/dags/provider_api_scripts/smithsonian.py [hunk 2]
```diff
@@ -291,7 +296,9 @@ def _get_title(row):
     return row.get('title')
 
 
-def _get_creator(row, creator_types=CREATOR_TYPES):
+def _get_creator(row, creator_types=None):
+    if creator_types is None:
+        creator_types = CREATOR_TYPES
```
[the remark this answers is not in the document]
Comment (obulat) [reply src/cc_catalog_airflow/dags/provider_api_scripts/smithsonian.py:305]: Thank you!

[the remark this answers is not in the document]
Comment (obulat) [reply src/cc_catalog_airflow/dags/provider_api_scripts/smithsonian.py]: I reviewed and added `.copy()` to all mutable default parameters.

**Review by krysal — DISMISSED** [review]
Great, thanks!

**Review by zackkrida — APPROVED** [review]
Nice

**Pull request merged**